src/backend/database_migrations/versions/20211222_115458_migrate_sample_locations_to_location_id.py

```diff
@@ -0,0 +1,37 @@
+"""Migrate sample locations to location_id
+
+Create Date: 2021-12-13 14:16:58.933547
+
+"""
+import enumtables  # noqa: F401
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "20211222_115458"
+down_revision = "20211214_003923"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+
+    set_full_locations_stmt = sa.sql.text(
+        "UPDATE aspen.samples as s SET location_id = (SELECT id FROM aspen.locations as l WHERE l.region = s.region AND l.country = s.country AND l.division = s.division and l.location = s.location)"
+    )
+    conn.execute(set_full_locations_stmt)
+
+    set_broader_locations_stmt = sa.sql.text(
+        "UPDATE aspen.samples as s SET location_id = (SELECT id FROM aspen.locations as l WHERE l.region = s.region AND l.country = s.country AND l.division = s.division and l.location is NULL) WHERE s.location IN ('California', 'NaN', '', NULL)"
+    )
+    conn.execute(set_broader_locations_stmt)
+
+    deprecated_columns = ["region", "country", "division", "location"]
+    with op.batch_alter_table("samples", schema="aspen") as batch_op:
+        for column in deprecated_columns:
+            batch_op.drop_column(column)
```

Comment:
How much validation have you done on the UPDATE queries above? I'm wondering if we should wait to actually drop the columns until we've validated that we have 0 samples with null locations?

Reply:
<img width="632" alt="Screen Shot 2022-01-03 at 09 58 22" src="https://user-images.githubusercontent.com/24234461/147980737-e96653f2-1113-4fa6-a29e-ead890668f00.png">